code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm

```diff
@@ -0,0 +1,111 @@
+///To be used when there is the need of an atmos connection without repathing everything (eg: cryo.dm)
+/obj/machinery/atmospherics/components/unary/machine_connector
+	name = "machine connector"
+	desc = "Internal connector to interface with the pipes."
+
+	can_unwrench = FALSE
+
+	use_power = NO_POWER_USE
+	layer = GAS_FILTER_LAYER
+	hide = TRUE
+	shift_underlay_only = FALSE
+
+	pipe_flags = PIPING_ONE_PER_TURF
+	resistance_flags = INDESTRUCTIBLE
+
+	///Reference to the machine we are connected to
+	var/obj/machinery/connected_machine
+
+/obj/machinery/atmospherics/components/unary/machine_connector/Initialize(mapload, process = TRUE, setdir, init_dir = ALL_CARDINALS, obj/machinery/connecting_machine, direction)
+	dir = direction
+	. = ..()
+	connected_machine = connecting_machine
+	if(!connected_machine)
+		return
+	SSair.start_processing_machine(connected_machine)
+	register_with_machine()
+
+/obj/machinery/atmospherics/components/unary/machine_connector/Destroy()
+	if(connected_machine)
+		connected_machine = null
+	return ..()
+
+/**
+ * Register various signals that are required for the proper work of the connector
+ */
+/obj/machinery/atmospherics/components/unary/machine_connector/proc/register_with_machine()
+	RegisterSignal(connected_machine, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(pre_move_connected_machine))
+	RegisterSignal(connected_machine, COMSIG_MOVABLE_MOVED, PROC_REF(moved_connected_machine))
+	RegisterSignal(connected_machine, COMSIG_MACHINERY_DEFAULT_ROTATE_WRENCH, PROC_REF(wrenched_connected_machine))
```

Comment:
wouldn't it be more robust to listen for `COMSIG_ATOM_DIR_CHANGE` instead?

Reply:
not really, machines shouldn't rotate without a wrench, so any changes to the dir that don't happen because of it are a potential issue

Reply:
what happens if they do? would you want to respect it or no

Reply:
Not really, machines aren't supposed to rotate without a wrench